tests/test_validate.py

```diff
@@ -107,3 +108,74 @@ def check(prurl):
     assert len(check(', https://flit.readthedocs.io/')) == 1
     # Name longer than 32 chars
     assert len(check('Supercalifragilisticexpialidocious, https://flit.readthedocs.io/')) == 1
+
+
+def test_read_classifiers_cached(monkeypatch):
+    def mock_get_cache_dir():
+
+        class MockChache:
+            def open(self, encoding):
+                return self
+            def __truediv__(self, other):
+                return self
+            def __enter__(self):
+                return ["A", "B", "C"]
+            def __exit__(self, *args):
+                pass
+
+        return MockChache()
+
+    monkeypatch.setattr(fv, "get_cache_dir", mock_get_cache_dir)
+
+    classifiers = fv._read_classifiers_cached()
+
+    assert classifiers == {'A', 'B', 'C'}
+
+
+def test_download_and_chache_classifiers():
+    classifiers = fv._download_and_chache_classifiers()
```

Comment:
In general, I like it to be possible to run the tests without requiring network access. There are two approaches for this:

1. Mark tests which need network access, so there's a clear way to exclude them. I don't think there are any examples in Flit, but here's one from another of my projects: https://github.com/takluyver/pynsist/blob/bf2408bbd01b5da63b93107f48435a5a57f9048b/nsist/tests/test_pypi.py#L11-L14
2. Mock out the network responses, e.g. see the `test_upload` module.

Reply:
Since you used `responses` already, I've choosen the second option.